include/SH3/system/glbuffer.hpp

```diff
@@ -0,0 +1,120 @@
+/** @file
+ *
+ *  A wrapper and more OO approach to OpenGL Vertex Buffer Objects. This is also helps prevent any
+ *  memory problems as well as providing a whole lot of helper functions for our VBO.
+ *
+ *  @copyright 2017 Palm Studios
+ *
+ *  @date 8-6-2017
+ *
+ *  @author Jesse Buhagiar
+ */
+#ifndef buffer_HPP_INCLUDED
+#define buffer_HPP_INCLUDED
+
+#include "GL/glew.h"
+#include "GL/gl.h"
+
+#include <string>
+#include <cstddef>
+
+namespace sh3_gl
+{
+    /**
+     *
+     * An OpenGL Vertex Buffer Object (VBO). Provides a nice, easy, typesafe way
+     * to create and manage OpenGL buffers.
+     */
+    struct sh3_buffer final
+    {
+    public:
+
+        enum class Target
+        {
+            ARRAY_BUFFER    = GL_ARRAY_BUFFER,
+            ELEMENT_BUFFER  = GL_ELEMENT_ARRAY_BUFFER,
+            TEXTURE_BUFFER  = GL_TEXTURE_BUFFER
+        };
+
+        /**
+         *  Default constructor.
+         */
+        sh3_buffer(Target type, const std::string& buffName = ""): buffType(type), name(buffName){Create();};
+
+        /**
+         *  Destructor. Ensure we have no leaks when this buffer is to be deleted by the C++ Runtime.
+         */
+        ~sh3_buffer(){Release();};
+
+        /**
+         *  Create our buffer and register it with OpenGL.
+         */
+        void Create();
+
+        /**
+         *  Free any data related to this buffer, instead of letting the destructor do it.
```

Comment:
I'm not too sure, however, see #94 This causes a segmentation fault in my test program, almost like the compiler is inserting the dtor to the colour buffer, which actually releases _during_ render and causing a segmentation violation.